userspace/libgolf2/aes_syscalls.c

```diff
@@ -0,0 +1,204 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "aes_ecb_syscalls.h"
+
+// Struct used for creating synchronous versions of functions.
+//
+// fired -  set when the callback has been called
+// error - error received from the kernel less than zero indicates an error
```

Comment:
Add a period between the sentences (after "kernel")

Reply:
I introduced tock_ to minimize namespace conflicts. In porting over this U2F app, I've realized that making libtock-c APIs *not* have tock_ in their names is problematic. For example, U2F defines function such as `sha256`. We can't have such functions defined by libtock.

Reply:
I see the need for namespacing, I'm just not sure about the namespace name. Perhaps "tock" is the right name, or perhaps "h1b" or "golf2" would be better.

Reply:
Fixed

Reply:
I think all system calls should be prefaced by tock_ since they're implemented in the kernel. But if a system call is platform or chip specific, that could be in the name as well (for clarity, not namespace reasons). E.g., `tock_h1b_keyladder`. In the case of AES, though, this API shouldn't be golf2/h1b-specific.